Entities/Items/Projectiles/Arrow.as

```diff
@@ -165,22 +163,40 @@ void onTick(CBlob@ this)
 
 	// sticking
 	if (processSticking)
-	{
-		//no collision
-		shape.getConsts().collidable = false;
-
-		if (!this.hasTag("_collisions"))
-		{
-			this.Tag("_collisions");
-			// make everyone recheck their collisions with me
-			const uint count = this.getTouchingCount();
-			for (uint step = 0; step < count; ++step)
+	{	
+		if (this.get_bool("hitBlobExists")) 
+		{
+			// structure blob: door, platform, team bridge, trap block etc.
+			
+			CBlob@ gottenBlob = getBlobByNetworkID(this.get_u32("hitBlob"));
+			
+			if (gottenBlob is null) // structure blob is gone
 			{
-				CBlob@ _blob = this.getTouchingByIndex(step);
-				_blob.getShape().checkCollisionsAgain = true;
+				this.server_Die();
+			}
+			else if (gottenBlob.hasTag("fallen")										// structure blob is falling
+					|| (gottenBlob.getName().find("door") != -1 && isOpen(gottenBlob))	// opening door
+					|| (gottenBlob.getName() == "bridge" && isOpen(gottenBlob))			// opening bridge
+					|| (gottenBlob.getName() == "trap_block" && isOpen(gottenBlob)))	// opening trap block
+			{
+				this.server_Die();
 			}
 		}
-
+		else 
+		{
+			// maptile: stone, wood, dirt, obstructor etc.
+			
+			CMap@ map = getMap();
+			Vec2f hitpos = this.get_Vec2f("hitWorldPoint");
+			
+			Tile hitTile = map.getTile(hitpos);
+			
+			if (!map.isTileSolid(hitTile))
+				this.server_Die();		
+		}
+		
```

Comment:
All of the code you added here (lines 166 to 196) should only be calculated by the server since all the pathways from the "hitBlobExists" check only do server_die() and nothing else. Add a ``isServer()`` check.